bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BatchedReadEntryProcessor.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.proto;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.util.Recycler;
+import io.netty.util.ReferenceCounted;
+import java.util.concurrent.ExecutorService;
+import org.apache.bookkeeper.proto.BookieProtocol.BatchedReadRequest;
+import org.apache.bookkeeper.util.ByteBufList;
+
+class BatchedReadEntryProcessor extends ReadEntryProcessor {
+
+    private long maxBatchReadSize;
+
+    public static BatchedReadEntryProcessor create(BatchedReadRequest request,
+                                            BookieRequestHandler requestHandler,
+                                            BookieRequestProcessor requestProcessor,
+                                            ExecutorService fenceThreadPool,
+                                            boolean throttleReadResponses,
+                                            long maxBatchReadSize) {
+        BatchedReadEntryProcessor rep = RECYCLER.get();
+        rep.init(request, requestHandler, requestProcessor);
+        rep.fenceThreadPool = fenceThreadPool;
+        rep.throttleReadResponses = throttleReadResponses;
+        rep.maxBatchReadSize = maxBatchReadSize;
+        requestProcessor.onReadRequestStart(requestHandler.ctx().channel());
+        return rep;
+    }
+
+    @Override
+    protected ReferenceCounted readData() throws Exception {
+        ByteBufList data = null;
+        BatchedReadRequest batchRequest = (BatchedReadRequest) request;
+        int maxCount = batchRequest.getMaxCount();
+        if (maxCount <= 0) {
+            maxCount = Integer.MAX_VALUE;
+        }
+        long maxSize = Math.min(batchRequest.getMaxSize(), maxBatchReadSize);
```

Comment:
For current logic, if the maxSize <0, it won't return any entries to the client and it will be hard to debug what happens. At least we need to add a debug level log.

Reply:
No, we will at least add one entry to the response whatever how size of the maxSize.
We just check the maxSize when the ByteBufList is not null.